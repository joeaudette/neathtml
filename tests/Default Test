			<div style="margin-top: -50px; height: 50px; background-color: red;">Let me out with a negative margin-top!</div>
			<p>Here is a table with lots of XSS attacks and tag soup markup:</p>
			<table border="1">
				<tr>
					<th>Script attacks
					<td>script element<br/>
						<script>
						window.alert('XSS from script element');
						</script>
						script elem in CDATA<br/>
						<![CDATA[
						<script>
						window.alert('XSS from script element in CDATA section');
						</script>
						]]>						
					<td>
						<a onclick=window.alert('XSS_on_click') href='http://www.google.com/'>on* attr</a><br />
						<a href=javascript:alert('XSS_on_link')>javascript href</a><br />
					<td id='styleXss' style='nonstandard-attribute1: expression(alert(&quot;XSS from style&quot;)); nonstandard-attribute2: expr/**/ession(alert(&quot;XSS from style with comment&quot;)); nonstandard-attribute3: expres\000073ion(alert(&quot;XSS from style with escape&quot;)); background-color: rgb(192,255,192);'>
						green despite script in style
					<td>
						<a id='trustedLinkBefore' href='http://www.google.com/'>spoof existing ID</a><br />
						<a id='trustedLinkAfter' href='http://www.google.com/'>spoof future ID</a><br />
				<tr>
					<th>Tag soup
					<td><ul><li>no &lt;/li> 1/2<li>no &lt;/li> 2/2</ul>
					<td>unmatched &lt;/em> </em>
					<td><B>varying</B> <i>case</I> <U>tags</u>
					<td><p>line<br>break with &lt;br&gt;
				<tr>
					<th>Special characters and attributes
					<td><a implicit_attr href=http://www.google.com/search?hl=en&q=neathtml&btnG=Search>unquoted and unencoded link</a>
					<td><font face='&#9;&#0;'>non-printable</font> and <font face='&#xdead;'>non-ascii</font> attr values
					<td>A B C with entities: &#65;&nbsp;&#x42;&nbsp;&#X43;
					<td>Unencoded <, and &
			</table>
			<br/>
			<table background="http://www.brettle.com/Data/Sites/1/logos/deanatwork_sidesmall.jpg" border="1" style="border-spacing: 0;">
				<tr>
					<td style="counter-increment: trusted-num;">Increment a CSS counter.  For result, see just under the
						untrusted content box.
					</td>
					<td id="innerTableContainer">
						Nested table:
						<table id="innerTable" border="1">
							<tr>
								<td style="background-image: url(http://www.brettle.com/Data/Sites/1/logos/deanatwork_sidesmall.jpg)">style="background-image: url(...)"</td>
								<td style="background-color: rgb(0,255,0);">style="background-color: rgb(...);"</td>
							</tr>
						</table>
					</td>
					<td>
						Try to break out of the layout jail
						<div style="position: absolute; top: 0; right: 0; color: red;">Let me out!</div>
						<div style="position: absolute; top: -100px; right: 0; color: red;">Let me out with a negative top property!</div>
					</td>
				</tr>
			</table>

			Try to break out of the markup jail...
			</table>
            </div>
            </div>
            </div>
			"Help! Let me out of this box!"
			Try to pull trusted content into the box...
			<iframe><object><script>

