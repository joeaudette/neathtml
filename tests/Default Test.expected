<div>			<div style="margin-top: -50px; height: 50px; background-color: red; ">Let me out with a negative margin-top!</div>
			<p>Here is a table with lots of XSS attacks and tag soup markup:</p>
			<table border="1">
				<tr>
					<th>Script attacks
					<td>script element<br />
						
						script elem in CDATA<br />
						
						&lt;script&gt;
						window.alert('XSS from script element in CDATA section');
						&lt;/script&gt;
												
					</td><td>
						<a href="http://www.google.com/">on* attr</a><br />
						<a>javascript href</a><br />
					</td><td id="NeatHtml_styleXss" style="background-color: rgb(192,255,192); ">
						green despite script in style
					</td><td>
						<a id="NeatHtml_trustedLinkBefore" href="http://www.google.com/">spoof existing ID</a><br />
						<a id="NeatHtml_trustedLinkAfter" href="http://www.google.com/">spoof future ID</a><br />
				<tr>
					<th>CSRF attacks
					<td>trusted image:<br />
						<img src="tests/trusted/test.png" />
					</td><td>untrusted images:<br />
						<img />
						<img />
						
					</td><td>SRC attr in &lt;script&gt;:<br />
						
					</td><td>SRC attr in &lt;iframe&gt;:<br />
						
				<tr>
					<th>Tag soup
					<td><ul><li>no &lt;/li> 1/2</li><li>no &lt;/li> 2/2</li></ul>
					</td><td>unmatched &lt;/em> 
					</td><td><B>varying</B> <i>case</i> <U>tags</U>
					</td><td><p>line<br />break with &lt;br&gt;
				<tr>
					<th>Special characters and attributes
					<td><a href="http://www.google.com/search?hl=en&amp;q=neathtml&amp;btnG=Search">unquoted and unencoded link</a>
					</td><td><font face="&#9;&#0;">non-printable</font> and <font face="&#57005;">non-ascii</font> attr values
					</td><td>A B C with entities: &#65;&#160;&#x42;&#160;&#x43;
					</td><td>Unencoded &lt;, and &amp;
			</td></th></tr></p></td></th></tr></td></th></tr></td></th></tr></table>
			<br />
			<table border="1" style="border-spacing: 0; ">
				<tr>
					<td style="">Increment a CSS counter.  For result, see just under the
						untrusted content box.
					</td>
					<td id="NeatHtml_innerTableContainer">
						Nested table:
						<table id="NeatHtml_innerTable" border="1">
							<tr>
								<td style="">style="background-image: url(...)"</td>
								<td style="background-color: rgb(0,255,0); ">style="background-color: rgb(...);"</td>
							</tr>
						</table>
					</td>
					<td>
						Try to break out of the layout jail
						<div style="position: absolute; top: 0; right: 0; color: red; ">Let me out!</div>
						<div style="position: absolute; top: -100px; right: 0; color: red; ">Let me out with a negative top property!</div>
					</td>
				</tr>
			</table>

			Try to break out of the markup jail...
			
            </div>
